docs/design/002-containerized-gadgets.md

```diff
@@ -0,0 +1,602 @@
+# Containerized Gadgets Support Inspektor Gadget
+
+The gadgets we provide are heavily coupled to Inspektor Gadget as they are all developed on the same
+repository and maintained by our team. Adding new gadgets implies that they have to be reviewed and
+approved by us. This approach presents some disadvantages:
+- Users need to install new versions of Inspektor Gadget to get new gadgets (or updates to them).
+- Too specific gadgets can't be added to our code base.
+- Users can't have private gadgets.
+
+The [containerized gadgets idea](https://github.com/inspektor-gadget/inspektor-gadget/issues/1669)
+aims to make Inspektor Gadget a framework to run eBPF programs (gadgets), like Docker does for
+containers. This approach decouples the gadgets from Inspektor Gadget solving the issues mentioned
+above.
+
+Many ideas of this design document are based on [bumblebee](https://github.com/solo-io/bumblebee)
+and [ebpf_exporter](https://github.com/cloudflare/ebpf_exporter) projects.
+## User Experience
+
+The UX we provide should be very close to the existing container runtimes like Docker.
+
+### `run`
+
+The run command starts a gadget:
+
+```bash
+$ ig run mygadgetimage:latest
+# print output to terminal
+```
+
+`mygadgetimage:latest` is gadget OCI image.
+This command should provide a flag to run on detached mode:
+
+```bash
+$ ig run mygadgetimage:latest --detached
+Created gadget foo
+```
+
+This command automatically pulls the image if it's available in the local image cache.
```

Comment:
Should this be "if it's _not_ in the local image cache". It should just run it if it is in the local image cache, right?

Reply:
Yes!